src/crud__operation.ts

```diff
@@ -0,0 +1,207 @@
+  //     sort: { rating: -1 },
+  //     // Include only the `title` and `imdb` fields in the returned document
+  //     projection: { _id: 0, title: 1, imdb: 1 },
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return collection.findOne(query);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+//Find all documents
+export const findAllUsers = async (query: object, options: object) => {
+  //Query
+  //const query = { runtime: { $lt: 15 } };
+  //const options = {
+  // sort returned documents in ascending order by title (A->Z)
+  //sort: { title: 1 },
+  // Include only the `title` and `imdb` fields in each returned document
+  //projection: { _id: 0, title: 1, imdb: 1 },
+  //};
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return collection.find(query, options);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Uodate one documents and update
+export const updateOneDoc = async (
+  filter: object,
+  options: object,
+  updateDoc: object
+) => {
+  //=====create a filter
+  //const filter = { title: "Blacksmith Scene" };
+  //=====this option instructs the method to create a document if no documents match the filter
+  //const options = { upsert: true };
+  //=====create a document
+  //const updateDoc = {
+  //     $set: {
+  //       plot:
+  //         "Blacksmith Scene is a silent film directed by William K.L. Dickson",
+  //     },
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.updateOne(filter, options, updateDoc);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Udate many documents
+export const updateManyDoc = async (filter: object, updateDoc: object) => {
+  //=====create a filter
+  //const filter = { rated: "G" };
+  //=====increment every document matching the filter with 2 more comments
+  // const updateDoc = {
+  //     $inc: {
+  //       num_mflix_comments: 2,
+  //     },
+  //   };
+  const client = new MongoClient(lockalConnection, dbOptions);
+  try {
+    await client.connect();
+    const database = client.db("test");
+    const collection = database.collection("users");
+    return await collection.updateOne(filter, updateDoc);
+  } catch (error) {
+    console.log(error);
+  } finally {
+    await client.close();
+  }
+};
+
+//Replace one document
+export const replaceOneDoc = async (
+  query: object,
+  replacement: object,
+  options: object
+) => {
+  //=====Query
```

Comment:
If this code is ready for review, please remove your comments. Comments should be used only if logic is super complex and we need to explain other developers behaviour

Reply:
This file will by deleted.